### PR TITLE
Use x crypto instead of sodium

### PR DESCRIPTION
**Pull request by andersjanmyr**
The example that is using libsodium is hard to cross compile.
Here's an alternative example using `google.com/x/crypto` instead.

Perhaps it should be added as an alternative file instead of change to the current one.

### CODE REVIEW

**Comment by gmlewis** [comment]
> Perhaps it should be added as an alternative file instead of change to the current one.

Yes, please add a new example and distinguish between the two with pros/cons in a comment if you wish.

The `libsodium` example was a special request to see how it might be done, so I'm totally fine with adding a new example.
Thank you, @andersjanmyr !

**Comment by andersjanmyr** [comment]
@gmlewis I pushed it as a new directory, but I didn't change the name of the old directory.
Perhaps I should change the name of it to `newreposecretwithlibsodium`?

File: example/newreposecretwithxcrypto/main.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
+// Copyright 2020 The go-github AUTHORS. All rights reserved.
```
Comment (gmlewis) [inline example/newreposecretwithxcrypto/main.go:1]:
```suggestion
// Copyright 2021 The go-github AUTHORS. All rights reserved.
```

File: example/newreposecretwithxcrypto/main.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	client := github.NewClient(tc)
+	return ctx, client, nil
+}
+
+// addRepoSecret will add a secret to a GitHub repo for use in GitHub Actions.
+//
+// Finally, the secretName and secretValue will determine the name of the secret added and it's corresponding value.
+//
+// The actual transmission of the secret value to GitHub using the api requires that the secret value is encrypted
+// using the public key of the target repo. This encryption must be done using sodium.
```
Comment (gmlewis) [inline example/newreposecretwithxcrypto/main.go:118]: > This encryption must be done using sodium.

Is this still true?

File: example/newreposecretwithxcrypto/main.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+//
+// First, the public key of the repo is retrieved. The public key comes base64
+// encoded, so it must be decoded prior to use in sodiumlib.
```
Comment (gmlewis) [inline example/newreposecretwithxcrypto/main.go:121]: > so it must be decoded prior to use in sodiumlib.

How about this?

File: example/newreposecretwithxcrypto/main.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+//
+// Second, the secret value is converted into a slice of bytes.
+//
+// Third, the secret is encrypted with sodium.CryptoBoxSeal using the repo's decoded public key.
```
Comment (gmlewis) [inline example/newreposecretwithxcrypto/main.go:125]: > Third, the secret is encrypted with sodium.CryptoBoxSeal using the repo's decoded public key.

...and this?

**Review by gmlewis — CHANGES_REQUESTED** [review]
> Perhaps I should change the name of it to `newreposecretwithlibsodium`?

Sure, that sounds like a good idea.

File: example/newreposecretwithxcrypto/main.go (hunk 2 above)
Comment (andersjanmyr) [reply example/newreposecretwithxcrypto/main.go:118]: You are right, I'll go through the comments and clean it up. Sorry about that.
\

File: example/newreposecretwithxcrypto/go.mod [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module newreposecret
```
Comment (gmlewis) [inline example/newreposecretwithxcrypto/go.mod:1]:
```suggestion
module newreposecretwithxcrypto
```

**Review by gmlewis — APPROVED** [review]
Thank you, @andersjanmyr !
LGTM.

If you want to rename the older libsodium example dir in this PR, that's fine with me.
Also, if you want to clean up any of the comment in that older example, that would be awesome too...  (I see that you cleaned up some of the wording here... and maybe some of those cleanups also apply to the older example, but if not, that's OK too... thanks!)

**Comment by andersjanmyr** [comment]
Fixed the comments and modules.

File: example/newreposecretwithlibsodium/main.go [hunk 6]
```diff
@@ -1,13 +1,13 @@
-// Copyright 2020 The go-github AUTHORS. All rights reserved.
+// Copyright 2021 The go-github AUTHORS. All rights reserved.
```
[the remark this answers is not in the document]
Comment (andersjanmyr) [reply example/newreposecretwithlibsodium/main.go]: 🤦 fixed.

**Review by gmlewis — APPROVED** [review]
Thank you, @andersjanmyr !
LGTM.
Merging.

**Pull request merged**